internal/internal_task_handlers.go

```diff
@@ -1623,27 +1623,63 @@ type cadenceInvoker struct {
 	heartBeatTimeoutInSec int32       // The heart beat interval configured for this activity.
 	hbBatchEndTimer       *time.Timer // Whether we started a batch of operations that need to be reported in the cycle. This gets started on a user call.
 	lastDetailsToReport   *[]byte
+	detailsToReport       *[]byte
 	closeCh               chan struct{}
 	workerStopChannel     <-chan struct{}
 }
 
-func (i *cadenceInvoker) Heartbeat(details []byte, skipBatching bool) error {
+func (i *cadenceInvoker) Heartbeat(details []byte) error {
```

Comment:
nit: probably we need to add some comment here saying `Heartbeat` should not be used together with the other two method as it may cause the progress to be lost?

Reply:
Yes will add comment.